extensions/common/sql/common-sql/src/main/java/org/eclipse/dataspaceconnector/sql/DoorKeeper.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+
+import java.util.Stack;
+
+/**
+ * The DoorKeeper takes care to close all the registered {@link AutoCloseable} components from the last to the first
+ * Wrapping eventually checked exceptions with an unchecked {@link EdcException}
+ */
+public class DoorKeeper implements AutoCloseable {
```

Comment:
minor nit: the `DoorKeeper` could be package-private, since it's only used by the `SqlQueryExecutor`. Or we could make it generally available, in which case I'd move it to one of the core packages.